File: leetcode/2270_Number_of_Ways_to_Split_Array/leetcode2270.cpp
#include<iostream>
#include<vector>
#include<numeric>
using namespace std;

class Solution{
    public:
    int waysToSplitArray(vector<int>&nums){
        int n = nums.size();
        int split_count = 0;
        long long total_sum = accumulate(nums.begin(), nums.end(), 0LL);
        long long left_sum = 0;
        for (int i = 0; i<n-1; i++){
            left_sum += nums.at(i);
            if (left_sum >= total_sum-left_sum) split_count += 1;
        }
        return split_count;
    }
};
int main(){
    Solution Sol;
    vector<int>vec={10,4,-8,7};
    cout<<Sol.waysToSplitArray(vec)<<endl;
}

File: leetcode/2270_Number_of_Ways_to_Split_Array/leetcode2270.py
from typing import List

class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        n = len(nums)
        valid_split = left_sum = 0
        total_sum = sum(nums)
        for i in range(n - 1):
            left_sum += nums[i]
            valid_split = (valid_split + 1 if left_sum >= total_sum - left_sum else valid_split)
        return valid_split

if __name__ == "__main__":
    sol = Solution()
    print(sol.waysToSplitArray([10, 4, -8, 7]))
    print(sol.waysToSplitArray([2, 3, 1, 0]))
